### DISCUSSION TITLE
Own fader at first position in the mixer

### DISCUSSION THREAD

**Discussion by ngocdh** [jamulussoftware/jamulus#1646] [UI Related]
Hi,

Own fader should be very easy to find, so I'm suggesting at least an option to always move own fader to first position in the mixer.

Further more, I'd like to request a feature: volume up or down the whole mix (by x percent for example). A very useful use case is when everyone is at 100% and I want my volume to be a bit higher than others in my mix, especially when I'm not the first to join the room.

Let me know what you think.
Cheers

**Comment by gilgongo** [comment]
Hi - the second of those is in the next release (to apply a gain in the settings): https://github.com/jamulussoftware/jamulus/pull/1222 The general feeling though is that device volume in general needs to be done outside of Jamulus, for various reasons. 

The "own fader position" is interesting. Wonder what others think? See also: https://github.com/jamulussoftware/jamulus/discussions/1639

**Reply by hoffie** [reply]
> The "own fader position" is interesting. Wonder what others think? See also: #1639

I understand that it is an issue, especially when having lots of participants and/or if there are name clashes.

When considering to implemen this, we should be clear how to handle several cases:
- Should this be always done or will it annoy users?
- How would this behave in combination with one of the sorting options?

**Reply by gilgongo** [reply]
This reminds me: I've often wondered why it is that we show your own fader in the mix at all. It's always felt a bit confusing to me given that you have what appears to be the same controls on the left of the screen as well.

**Reply by hoffie** [reply]
> given that you have what appears to be the same controls on the left of the screen as well.

Well, at least mute-myself and one's own mixer level would be missing, wouldn't they?

**Reply by ngocdh** [reply]
Hi @hoffie 
> - Should this be always done or will it annoy users?

I'd suggest a checkbox, something like "Move own fader to the left", maybe in the Edit menu, close to the sorting options. It doesn't have to be enabled by default. When users get used to the feature, maybe they will prefer to enable it by default.

> - How would this behave in combination with one of the sorting options?

If the option is off, sorting works just like it does now.
If the option is on, then own fader is always the first one. If user belongs to a group, then the group sticks to the user: they become the first group. If user doesn't want this, then they just have to disable the option.

**Reply by gilgongo** [reply]
>  at least mute-myself and one's own mixer level would be missing,

I doubt anyone would naturally understand the difference between myself and mute on their own channel. Similarly with fader on their own channel and input level on the left of the window. They seem oddly similar...

**Reply by ngocdh** [reply]
@gilgongo tooltips?

**Reply by hoffie** [reply]
I totally agree that it is confusingfor newcomers, I just wanted to point out that they have different functionality and I think both features are relevant.

**Reply by gilgongo** [reply]
> @gilgongo tooltips?

Not on mobile/tablet! :-)

**Reply by gilgongo** [reply]
> I think both features are relevant.

But would anyone mind if we removed your own fader from the mix, made "mute myself" say "mute" (and maybe added a "solo" button) while making your main fader adjust the volume to the server?  What is the utility in being able to "mute myself" as well as mute my channel, or turn myself up as well as down?

**Reply by hoffie** [reply]
> What is the utility in being able to "mute myself" as well as mute my channel

I frequently mute myself in the mix when talking (and obviously do not mute my input) as I find any echo or delay highly confusing when speaking, but will want to hear myself (with delay) during singing, of course.

> or turn myself up as well as down

Not sure which part you mean -- there is a fader for my own level in the mix, but there is no input volume fader unless I'm missing something?

I think it's useful to visualize oneself as part of the mix (as it is now). Maybe we could do better highlighting this entry (by different positioning as suggested here and/or making this entry stand out in another way).

**Reply by gilgongo** [reply]
The distinction between "mute myself so that my signal no longer goes to the server" is separate from "mute myself so that it does go to the server but that only I cannot hear myself" is separate but maybe only useful in the one scenario you give, I agree. Whether we have to labour a convoluted "mixing desk" analogy to do that is the question, I think.

By volume, in mean you can adjust your device volume outside of Jamulus, as well as inside it in your mix (and as of the next release, inside settings for gain too).

**Reply by dcorson-ticino-com** [reply]
I use "mute myself so that my signal no longer goes to the server" every day, but can't remember ever using the mute on my channel, although I can imagine hoffie's use case on a high delay server.

**Reply by ngocdh** [reply]
I use both Mute buttons when I'm in Listener mode: "Mute myself" to avoid bothering people, and "Mute" my fader so that the mix does not include my background noise.

**Reply by henkdegroot** [reply]
I always have my own channel on mute. With my large instrument (tuba) it is not possible to ignore the sound from my own instrument and hearing it with echo/delay in my headset is very annoying. So I would at least like to have the option to mute my own channel.

**Reply by gilgongo** [reply]
Yes, I've no doubt these features are useful, but what I'm saying is that if the concept of "you" is **already** on the left in the main window (if not the mixer window), then why should it be necessary to see your fader in the mix as well? 

This has been a source of confusion to new people, but it's been very hard for them to articulate why, because Jamulus presents itself as a being a mixing desk. So to want to **not** have your fader in the mix would be in basic violation of that idea. But a mixing desk isn't a very good analogy for what's actually happening to your signal. So when it comes to the concept of "Mute Myself", since you can mute yourself in more than one place, people are naturally confused. Witness also the calls for an input level control too, which I think may be caught up in this confusion. I doubt anyone would have asked for it if your slider was simply on the left next to your input levels.

**Reply by jp8** [reply]
> Jamulus presents itself as a being a mixing desk. So to want to **not** have your fader in the mix would be in basic violation of that idea. But a mixing desk isn't a very good analogy for what's actually happening to your signa

Really, the Jamulus client is two separate things:
1. A software that collects audio from your microphone, and sends it to the mixer (server)
2. A software that allows you to create a 'mix' on the mixer (server) and hear that mix through your headphones.

Muting is possible in both part 1 and part 2.  Perhaps the user interface would benefit from a very solid line that separates these two functions.

**Reply by pljones** [reply]
It's meant to.  Everything on the left is meant to be local.  Everything on the right is meant to be server.

Except for the bottom left corner where a bunch of random bits appear.

**Comment by ngocdh** [comment]
> Hi - the second of those is in the next release (to apply a gain in the settings): https://github.com/jamulussoftware/jamulus/pull/1222 The general feeling though is that device volume in general needs to be done outside of Jamulus, for various reasons. 

Hi @gilgongo it's not really the same thing. When I'm already loud and just want to lower others volume, input gain won't help. The requested feature is similar to the Volume Mixer in Windows where you can lower everything down proportionally, and then up the fader you want.

> The "own fader position" is interesting. Wonder what others think? See also: https://github.com/jamulussoftware/jamulus/discussions/1639
> 

Yes initially I wanted to comment there, but thought maybe this should be requested separately.

**Reply by gilgongo** [reply]
> you can lower everything down proportionally, and then up the fader you want.

Could you use Jamulus's grouping for that?

**Reply by ngocdh** [reply]
> Could you use Jamulus's grouping for that?

Good point, but do we have a way to quickly put everyone in a group?

**Reply by gilgongo** [reply]
I suppose the current system could be improved?

**Reply by jp8** [reply]
> I suppose the current system could be improved?

I think I would like to propose merging the multi-row and the grouping functionality....  so that by definition each group is also a row of faders.

Then I would allow group (row) names rather than the current group numbering scheme.  And I would allow selection of (multiple) individual faders, but also of (multiple) entire rows.  And actions to mute/solo/adjust the selected faders.

**Comment by ghost** [comment]
If there are several sort criteria, they should be prioritised by the user
in the user settings.
eg. sort A third, sort B first, sort C none, sort D Nth sort E none

No sort check box should be on the main menu.
eg. the DIR command in MSDOS can be configured perfectly in
autoexec.bat or config.sys to perfect the directory listing.
(it appears modern operating systems have lost this ability)

**Comment by pgScorpio** [comment]
> Hi,
> 
> Own fader should be very easy to find, so I'm suggesting at least an option to always move own fader to first position in the mixer.
> 
> Further more, I'd like to request a feature: volume up or down the whole mix (by x percent for example). A very useful use case is when everyone is at 100% and I want my volume to be a bit higher than others in my mix, especially when I'm not the first to join the room.
> 
> Let me know what you think. Cheers

Own fader always at first position would conflict with all sorting options! 
Nevertheless I think it's a good idea to make your own fader easier to find i.e. by a colored border or a different knob color.

Volume up/down the whole mix can already be done by assigning everybody to a single group. or using "set all faders to new client level"
